File: csharp/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private const double EarthSecondsPerYear = 31557600;

    private readonly long _seconds;

    public SpaceAge(long seconds)
    {
        _seconds = seconds;
    }

    public double OnEarth() => CalculateAge(1.0);

    public double OnMercury() => CalculateAge(0.2408467);

    public double OnVenus() => CalculateAge(0.61519726);

    public double OnMars() => CalculateAge(1.8808158);

    public double OnJupiter() => CalculateAge(11.862615);

    public double OnSaturn() => CalculateAge(29.447498);

    public double OnUranus() => CalculateAge(84.016846);

    public double OnNeptune() => CalculateAge(164.79132);

    private double CalculateAge(double factor) => 
        Math.Round(_seconds / (EarthSecondsPerYear * factor), 2);
}

File: typescript/space-age/space-age.ts
class SpaceAge {
  private _seconds: number

  constructor(seconds: number) {
    this._seconds = seconds
  }

  get seconds(): number {
    return this._seconds
  }

  public onEarth(): number {
    return this.calculateAge(1.0)
  }

  public onMercury(): number {
    return this.calculateAge(0.2408467)
  }

  public onVenus(): number {
    return this.calculateAge(0.61519726)
  }

  public onMars(): number {
    return this.calculateAge(1.8808158)
  }

  public onJupiter(): number {
    return this.calculateAge(11.862615)
  }

  public onSaturn(): number {
    return this.calculateAge(29.447498)
  }

  public onUranus(): number {
    return this.calculateAge(84.016846)
  }

  public onNeptune(): number {
    return this.calculateAge(164.79132)
  }

  private calculateAge(factor: number): number {
    const age = this._seconds / (31557600 * factor)
    return Math.round(age * 100) / 100
  }
}

export default SpaceAge
